Import WPT tests for referenceTarget.

alice opened the pull request:
#### 1e2913a16d0da178b8e8ced292384b972d3f64f3
<pre>
Import WPT tests for referenceTarget.

<a href="https://bugs.webkit.org/show_bug.cgi?id=286806">https://bugs.webkit.org/show_bug.cgi?id=286806</a>

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/adf401057131b5b0c9afb04d6d4f93a60aacea0f">https://github.com/web-platform-tests/wpt/commit/adf401057131b5b0c9afb04d6d4f93a60aacea0f</a>

Reviewed by Manuel Rego Casasnovas.

* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/aria-labelledby-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/aria-labelledby.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/dom-mutation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/dom-mutation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-descendant-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-descendant.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-for-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-for.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/popovertarget-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/popovertarget.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/property-reflection-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/property-reflection.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/reference-target-basics-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/reference-target-basics.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/w3c-import.log: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.

Canonical link: <a href="https://commits.webkit.org/290165@main">https://commits.webkit.org/290165@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/932e083a9ba6db8ad7db81a2a6d64ccad9b9b662

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89072 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8596 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/43967 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94048 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/39833 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/8983 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/16782 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68626 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26303 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92074 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/6869 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/80905 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/48988 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6618 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35282 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/38940 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/76980 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/36269 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/95884 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16252 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/11948 "Found 2 new test failures: fast/dom/crash-with-bad-url.html imported/w3c/web-platform-tests/fullscreen/api/element-request-fullscreen-namespaces.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77504 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16508 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/76693 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/76792 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21191 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/19871 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/9344 "The change is no longer eligible for processing.") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/13971 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16266 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16007 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19458 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/17788 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

keithamus commented on LayoutTests/TestExpectations:
```diff
@@ -7602,6 +7602,9 @@ imported/w3c/web-platform-tests/css/css-anchor-position/position-visibility-no-o
 
+# anchor HTML attribute
+imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html [ Failure ]
+
```
This line shouldn't be necessary I think, because the `anchor-expected.txt` should be sufficient here.

alice replied:
It's not strictly necessary now that I've added the platform-specific expectations, but the extra failures on those platforms made me want to add it - the failures are specific about positions which are only the way they are because the attribute is not yet implemented. Ideally we shouldn't need platform-specific expectations for this test at all, I think.

keithamus commented on LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/property-reflection-expected.txt:
```diff
@@ -0,0 +1,4156 @@
+
+PASS button.ariaControlsElements has reflection behavior ReflectsHostInArray when pointing to button with reference targetappendTestDeclaratively
```
From the build failures it looks like this file needs updating.

alice replied:
Done.

mrego reviewed (approved):
r+

Commit:
Import WPT tests for referenceTarget.

https://bugs.webkit.org/show_bug.cgi?id=286806

Upstream commit: https://github.com/web-platform-tests/wpt/commit/adf401057131b5b0c9afb04d6d4f93a60aacea0f

Reviewed by Manuel Rego Casasnovas.

* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/README.md: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/aria-labelledby-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/aria-labelledby.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/dom-mutation-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/dom-mutation.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-descendant-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-descendant.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-for-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/label-for.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/popovertarget-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/popovertarget.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/property-reflection-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/property-reflection.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/reference-target-basics-expected.txt: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/reference-target-basics.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/w3c-import.log: Added.
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/reference-target/tentative/anchor-expected.txt: Added.

Canonical link: https://commits.webkit.org/290165@main
